src/server/service/kv.rs

```diff
@@ -2091,7 +2091,12 @@ txn_command_future!(future_acquire_pessimistic_lock, PessimisticLockRequest, Pes
                 tracker.write_write_detail(resp.mut_exec_details_v2().mut_write_detail());
             });
         },
-        Err(e) | Ok(Err(e)) => resp.set_errors(vec![extract_key_error(&e)].into()),
+        Err(e) | Ok(Err(e)) =>  {
+            GLOBAL_TRACKERS.with_tracker(tracker, |tracker| {
```

Comment:
What about just moving L2089 out of match? There can be `scan_detail` even if it fails to lock.

Reply:
done